lib/pure/streams.nim

```diff
@@ -135,6 +135,12 @@ type
     flushImpl*: proc (s: Stream)
       {.nimcall, raises: [Defect, IOError, OSError], tags: [WriteIOEffect], gcsafe.}
 
+  StreamOption* = enum  ## What operation the stream can do.
+    soReadable,         ## You can read data from stream.
+    soWritable,         ## Data can be written to the stream.
+    soSeekable,         ## Read or write position can be changed.
+    soPeekable          ## Can read data without advancing read position.
+
```

Comment:
Why is this required? Shouldn't the new asserts be good enough?

Reply:
`proc newFileStream*(f: File)` is called from several procedures and it always set procedure pointers in stream even if it must not be called.
These procedure pointers need to be set to nil to assert check works.
`newFileStream` and `openFileStream` proc with filename parameter and `inputStream`/`outputStream`/`errorStream` procs in osproc module except windows call `proc newFileStream*(f: File)`.

`StreamOption` allows these procs to ask `proc newFileStream*(f: File; options: set[StreamOption])` to set invalid procedure pointers to nil.
It can also be used for creating a stream from unseekable file descripter (pipe, socket, FIFO, etc) that raise assertion when `setPosition`/`getPosition` is called.
Should these procs set invalid procedure pointers to nil after calling `proc newFileStream*(f: File)` instead of adding `options: set[StreamOption]` parameter to it?